### DISCUSSION TITLE
Initial notes

### DISCUSSION THREAD

**Discussion by greenc-FNAL** [FNALssi/spack-localbuildcache#1] [General]
1. Prefer to invoke `spack buildcache create` via Python function call.
2. Can this functionality be generalized and subsumed into `spack buildcache create`?
3. Suggest adding user-configurable output build cache location (including—where appropriate—an existing configured mirror).

**Comment by marcmengel** [comment]
Item 1., above,  now completed.  3x speedup in my tests of buildcache-creating an environment with py-black and python deps.  
 
Item 2 -- this *could* be subsumed into spack buildcache create.  The code is largely stolen from it.  The check for whether things
      are from upstream or not would have to be reworked though, as it currently only works for environments.   It would be better
      to go learn from spack.spec.Spec.install_status how to tell whats upstream or not ('[+]' vs '[^]' ) and work that into 
      spack buildcache create.

item 3 -- coming  soon...
